### ISSUE TITLE
Useless parentheses: false positive with assignment

### ISSUE THREAD

**Issue by greg0ire** [slevomat/coding-standard#1066]
The rule `SlevomatCodingStandard.PHP.UselessParentheses` breaks the following piece of code like so:

```diff
-            'has_constructor'  => (null !== $constructor = $class->getConstructor()) && $constructor->getNumberOfParameters() > 0,
+            'has_constructor'  => null !== $constructor = $class->getConstructor() && $constructor->getNumberOfParameters() > 0,
```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/4f5ffcf7b7ae02a80fe06969a3c5763058a51b49

**Comment by greg0ire** [comment]
Thank you !
